### PR TITLE
Update synchronizer to typescript - Closes #5192

### LINKED ISSUE #5192: Update application/node/synchronizer to be typescript
### Description

Update framework application/node/synchronizer to be typescript

### Motivation

We gradually need to convert all the framework code into typescript

### Acceptance Criteria

- All files under framework/src/application/node/synchronizer should be in typescript
- All related test should be in typescript

### PR DESCRIPTION
**Pull request by shuse2**
### What was the problem?

This PR resolves #5192 

### How was it solved?

- Update synchronizer to typescript

### How was it tested?

- Check all eslint issue is resolved for synchronizer
- Check all tests are passing

### CODE REVIEW

File: framework/src/application/network/network.ts [hunk 1]
```diff
@@ -310,7 +310,9 @@ export class Network {
 				? request.procedure
 				: `app:${request.procedure}`;
 			try {
-				const result = await this.channel.invokePublic(sanitizedProcedure, {
+				const result = await this.channel.invokePublic<
+					liskP2p.p2pTypes.P2PNodeInfo
```
Comment (ishantiw) [inline framework/src/application/network/network.ts:314]: I know its not part of this PR but if we can change `liskP2p` -> `liskP2P`

File: framework/src/application/node/synchronizer/block_synchronization_mechanism.ts [hunk 2]
```diff
@@ -162,12 +199,18 @@ class BlockSynchronizationMechanism extends BaseSynchronizer {
 	 * preference over the current tip. If so, the temporary chain is restored
 	 * on top of the current chain and the blocks temp table is cleaned up
 	 */
-	async _handleBlockProcessingError(lastCommonBlock, peerId) {
+	private async _handleBlockProcessingError(
+		lastCommonBlock: BlockInstance,
+		peerId: string,
+	) {
```
Comment (ManuGowda) [inline framework/src/application/node/synchronizer/block_synchronization_mechanism.ts:205]: return type

File: framework/src/application/node/synchronizer/block_synchronization_mechanism.ts [hunk 3]
```diff
@@ -238,10 +281,10 @@ class BlockSynchronizationMechanism extends BaseSynchronizer {
...
 	}
 
-	async _requestAndApplyBlocksToCurrentChain(
-		receivedBlock,
-		lastCommonBlock,
-		peerId,
+	private async _requestAndApplyBlocksToCurrentChain(
+		receivedBlock: BlockInstance,
+		lastCommonBlock: BlockInstance,
+		peerId: string,
```
Comment (ManuGowda) [inline framework/src/application/node/synchronizer/block_synchronization_mechanism.ts:287]: return type missing

File: framework/src/application/node/synchronizer/block_synchronization_mechanism.ts [hunk 4]
```diff
@@ -282,7 +325,7 @@ class BlockSynchronizationMechanism extends BaseSynchronizer {
 		return true;
 	}
 
-	async _revertToLastCommonBlock(peerId) {
+	private async _revertToLastCommonBlock(peerId: string) {
```
Comment (ManuGowda) [inline framework/src/application/node/synchronizer/block_synchronization_mechanism.ts:328]: same here, and below ones have missing return type too
